### ISSUE TITLE
Non-chronological line chart

### ISSUE THREAD

**Issue by Nicolas469** [naver/billboard.js#2420]
## Description
When giving a non-chronological dataset, the line chart will re-order the points before drawing lines.

[Exemple](https://stackblitz.com/edit/de9awd-yr6opd?file=index.ts) where x are given in non-chronological order: 3, 1, 4, 2.

## Desired output
Something like [this](https://stackblitz.com/edit/de9awd?file=index.ts). To get this result I've decompose the function in 3 segments and made some tweaks in the legend to give the impression of only one line.

**Comment by netil** [comment]
Hi @Nicolas469, I think non-chronological can be achieved by setting x axis type as `category`.

working example:
- https://stackblitz.com/edit/de9awd-nzrjx4

```js
bb.generate({
  data: {
    x: 'x1',
    columns: [
      ['x1', 3, 1, 4, 2],
      ['d1', 4, 3, 2, 1],
    ],
    type: 'line',
    colors: {
      d1: '#000000'
    }
  },
  axis: {
    x: {
      type: 'category',
      tick: {
        centered: true
      }
    }
  }
})
```

**Comment by Nicolas469** [comment]
Hi, thank you for your answer but it's not really what I want.
I would like the x axis to be chronologic but with data given in a non-chonological order.
If it's not clear, please check my example in the original question.
Thanks

**Comment by netil** [comment]
with rotated axis & category type, can achieve easily.

<img src=https://user-images.githubusercontent.com/2178435/145966796-4827db12-f280-45b6-a412-cef873f846e0.png width=300>


```js
bb.generate({
  data: {
    x: 'x1',
    columns: [
      ['x1', 4, 3, 2, 1],
      ['d1', 3, 1, 4, 2],
    ],
    type: 'line',
    colors: {
      d1: '#000000',
    },
  },
  axis: {
    rotated: true,
    x: {
      type: 'category',
    },
  },
});
```

**Comment by Nicolas469** [comment]
Yes I know that in this particular case it's possible.
But I'd like a more general solution.

**Comment by Nicolas469** [comment]
Yes it's working in this particular case. But I'd like a more general solution: defined the line as a "path".
Example (with Hightcharts): https://jsfiddle.net/ebmc1z9o/
I defined the data as a tuple: [[0, 15], [10, -50], [5, 10], [3, -30], [20, -56.5], [16, 20], [12, -25]]
With the same data, the output here is this: https://stackblitz.com/edit/de9awd-yr6opd?file=index.ts
